src/Raven.Server/Documents/Handlers/Processors/AbstractHandlerProcessor.cs

```diff
@@ -43,10 +45,14 @@ internal abstract class AbstractDatabaseHandlerProcessor<TRequestHandler, TOpera
 
         protected readonly Logger Logger;
 
+        public AuthenticateConnection AuthenticateConnection;
+
+
         protected AbstractDatabaseHandlerProcessor([NotNull] TRequestHandler requestHandler) : base(requestHandler)
         {
             ContextPool = requestHandler.ContextPool;
             Logger = requestHandler.Logger;
+            AuthenticateConnection = HttpContext.Features.Get<IHttpAuthenticationFeature>() as AuthenticateConnection;
```

Comment:
IIRC, this is not free (this is an IEnumerable) and you are now performing that for every single call to the database, even if this is not needed

Reply:
Done